File: src/main/java/com/rsupport/download/FileDownloader.java
package com.rsupport.download;

import com.rsupport.download.google.drive.GoogleDriveDownload;

import java.io.File;

public class FileDownloader {

    public static File download(String credentialFilePath, String fileId, String fileName) {
        return new GoogleDriveDownload(credentialFilePath, fileId, fileName).execute();
    }
}

File: src/main/java/com/rsupport/download/FileDownloader.kt
package com.rsupport.download

import com.google.api.client.auth.oauth2.Credential
import com.rsupport.google.drive.GoogleDriveDownload
import java.io.File

object FileDownloader {
    fun download(credential: Credential?, spreadSheetFieldId: String?, outputXlsxFilePath: String): File? {
        if(credential == null) throw NullPointerException("Credential is null.")
        if(spreadSheetFieldId.isNullOrEmpty()) throw NullPointerException("Spread Sheet Id is null.")
        val downloadable = GoogleDriveDownload(credential, spreadSheetFieldId, outputXlsxFilePath)
        return downloadable.execute()
    }
}